### PR TITLE
Button secondary hover constants

### PR DESCRIPTION
**Pull request by AnaMontes11**
Ready to review "ButtonSecondary hover" (default and inverse) Dark & Light Mode.
-Movistar
-O2
-O2 classic
-Vivo
-Telefónica
-Blau
-Solar360

### ------Update 30/11/22-----

* Usage of neutrals for secondary `hover` and `selected` states
* Improvements in link `hover` and `selected` states color contrast

Example: 
https://codepen.io/AlexBueno/pen/jOKLyMv

### CODE REVIEW

File: tokens/movistar.json [hunk 1]
```diff
@@ -125,27 +125,55 @@
...
+      "description": "movistarBlue"
+    },
+
+    "buttonSecondaryBackgroundSelectedInverse:": {
+      "value": "rgba({palette.movistarBlue}, 0.2)",
+      "type": "color",
+      "description": "movistarBlue"
+    },
+
+    "ButtonPrimary": {
```
Comment (yceballost) [inline tokens/movistar.json]: what is `ButtonPrimary`?

maybe `textButtonPrimary`?

**Review by yceballost — CHANGES_REQUESTED** [review]

**Comment by yceballost** [comment]
what about Solar360?

File: tokens/schema/skin-schema.json [hunk 2]
```diff
@@ -62,10 +62,14 @@
         "buttonPrimaryBackgroundSelected",
         "buttonPrimaryBackgroundSelectedInverse",
         "buttonPrimaryBackgroundHover",
-        "buttonSecondaryBackground",
+        "buttonSecondaryBorder",
```
Comment (yceballost) [inline tokens/schema/skin-schema.json:65]: well done! but you have to include the rest of the new constants :)

**Review by yceballost — CHANGES_REQUESTED** [review]

File: tokens/movistar.json [hunk 3]
```diff
@@ -125,26 +125,54 @@
...
       "description": "white"
     },
+
     "buttonSecondaryBorderSelectedInverse": {
       "value": "{palette.movistarBlue30}",
       "type": "color",
       "description": "movistarBlue30"
     },
+
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/movistar.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
you have a typo

File: tokens/o2.json [hunk 4]
```diff
@@ -552,25 +544,45 @@
...
       "description": "darkModeO2BluePrimary"
     },
     "buttonSecondaryBorderSelectedInverse": {
-      "value": "{palette.darkModeO2BluePrimaryDark}",
+      "value": "{palette.o2BluePrimary30}",
       "type": "color",
-      "description": "darkModeO2BluePrimaryDark"
+      "description": "o2BluePrimary30"
+    },
+    "buttonSecondaryBackgroundhover": {
```
Comment (yceballost) [inline tokens/o2.json]:
```suggestion
    "buttonSecondaryBackgroundHover": {
```
Typo here in H

File: tokens/o2.json [hunk 5]
```diff
@@ -552,25 +544,45 @@
       "type": "color",
       "description": "darkModeO2BluePrimaryDark"
     },
```
Comment (yceballost) [inline tokens/o2.json:574]: missing tones of constant here!
image

File: tokens/o2.json [hunk 6]
```diff
@@ -125,61 +125,53 @@
...
+    "buttonSecondaryBorderSelectedInverse": {
+      "value": "{palette.o2BluePrimary30}",
       "type": "color",
-      "description": "o2BluePrimary"
+      "description": "o2BluePrimary30"
     },
-    "textButtonPrimaryInverseSelected": {
-      "value": "{palette.o2BluePrimary70}",
+
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/o2.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/o2.json [hunk 7]
```diff
@@ -125,61 +125,53 @@
...
       "description": "o2BluePrimary70"
     },
-    "textButtonSecondary": {
-      "value": "{palette.o2BluePrimary}",
-      "type": "color",
-      "description": "o2BluePrimary"
-    },
-    "textButtonSecondarySelected": {
-      "value": "{palette.o2BluePrimary70}",
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/o2.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/solar-360.json [hunk 8]
```diff
@@ -145,6 +155,16 @@
       "type": "color",
       "description": "repsolOrange20"
     },
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/solar-360.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/solar-360.json [hunk 9]
```diff
@@ -145,6 +155,16 @@
...
+      "value": "rgba({palette.repsolOrange70}, 0.2)",
+      "type": "color",
+      "description": "repsolOrange70"
+    },
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/solar-360.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/solar-360.json [hunk 10]
```diff
@@ -614,25 +634,45 @@
...
+      "description": "white"
     },
     "buttonSecondaryBorderSelectedInverse": {
-      "value": "{palette.repsolOrange700}",
+      "value": "{palette.repsolOrange20}",
       "type": "color",
-      "description": "repsolOrange700"
+      "description": "repsolOrange20"
+    },
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/solar-360.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/solar-360.json [hunk 11]
```diff
@@ -614,25 +634,45 @@
...
+      "value": "rgba({palette.repsolOrange70}, 0.2)",
+      "type": "color",
+      "description": "repsolOrange70"
+    },
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/solar-360.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/telefonica.json [hunk 12]
```diff
@@ -125,25 +125,45 @@
...
       "description": "white"
     },
     "buttonSecondaryBorderSelectedInverse": {
-      "value": "{palette.telefonicaBlue30}",
+      "value": "{palette.telefonicaBlue20}",
       "type": "color",
-      "description": "telefonicaBlue30"
+      "description": "telefonicaBlue20"
+    },
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/telefonica.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/telefonica.json [hunk 13]
```diff
@@ -125,25 +125,45 @@
...
+      "value": "rgba({palette.telefonicaBlue70}, 0.2)",
+      "type": "color",
+      "description": "telefonicaBlue70"
+    },
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/telefonica.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/telefonica.json [hunk 14]
```diff
@@ -552,12 +572,22 @@
...
-      "description": "telefonicaBlue"
+      "description": "telefonicaBlue70"
     },
-    "buttonSecondaryBackgroundSelected": {
+    "buttonSecondaryBorder": {
+      "value": "{palette.white}",
+      "type": "color",
+      "description": "white"
+    },
+    " buttonSecondaryBackgroundHover": {
```
Comment (yceballost) [inline tokens/telefonica.json]:
```suggestion
    "buttonSecondaryBackgroundHover": {
```
you have a space at the beginning of this! these space cause problems, remove it!

File: tokens/vivo.json [hunk 15]
```diff
@@ -572,6 +609,16 @@
       "type": "color",
       "description": "vivoPurpleDark"
     },
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/vivo.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/vivo.json [hunk 16]
```diff
@@ -572,6 +609,16 @@
...
+      "value": "rgba({palette.vivoPurpleLight50}, 0.2)",
+      "type": "color",
+      "description": "vivoPurpleLight50"
+    },
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/vivo.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse:": {
```
same typo here

File: tokens/o2-classic.json [hunk 17]
```diff
@@ -125,26 +125,53 @@
...
       "description": "white"
     },
+
     "buttonSecondaryBorderSelectedInverse": {
       "value": "{palette.o2SkyBlueLight45}",
       "type": "color",
       "description": "o2SkyBlueLight45"
     },
+
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/o2-classic.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/o2-classic.json [hunk 18]
```diff
@@ -125,26 +125,53 @@
...
+      "value": "rgba({palette.o2SkyBlueLight45}, 0.2)",
+      "type": "color",
+      "description": "o2SkyBlueLight45"
+    },
+
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/o2-classic.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/movistar.json [hunk 19]
```diff
@@ -125,26 +125,54 @@
...
+      "type": "color",
+      "description": "movistarBlue"
+    },
+
+    "buttonSecondaryBackgroundSelectedInverse:": {
+      "value": "rgba({palette.movistarBlue}, 0.2)",
+      "type": "color",
+      "description": "movistarBlue"
+    },
+
```
Comment (yceballost) [inline tokens/movistar.json]:
```suggestion
```
remove these spaces pls

File: tokens/movistar.json [hunk 20]
```diff
@@ -125,26 +125,54 @@
...
       "description": "movistarBlue30"
     },
+
+    "buttonSecondaryBackgroundHoverInverse:": {
+      "value": "rgba({palette.movistarBlue}, 0.2)",
+      "type": "color",
+      "description": "movistarBlue"
+    },
+
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/movistar.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

File: tokens/movistar.json [hunk 21]
```diff
@@ -125,26 +125,54 @@
...
       "type": "color",
       "description": "white"
     },
+
     "buttonSecondaryBorderSelectedInverse": {
       "value": "{palette.movistarBlue30}",
       "type": "color",
       "description": "movistarBlue30"
     },
+
```
Comment (yceballost) [inline tokens/movistar.json]:
```suggestion
```
remove these spaces pls!

File: tokens/movistar.json [hunk 22]
```diff
@@ -125,26 +125,54 @@
...
+      "type": "color",
+      "description": "movistarBlue10"
+    },
+
     "buttonSecondaryBorderInverse": {
       "value": "{palette.white}",
       "type": "color",
       "description": "white"
     },
+
```
Comment (yceballost) [inline tokens/movistar.json]: remove these spaces pls!

File: tokens/movistar.json [hunk 23]
```diff
@@ -125,26 +125,54 @@
...
       "description": "movistarBlue"
     },
-    "buttonSecondaryBackgroundSelected": {
+  
+    "buttonSecondaryBorderSelected": {
       "value": "{palette.movistarBlue55}",
       "type": "color",
       "description": "movistarBlue55"
     },
+
```
Comment (yceballost) [inline tokens/movistar.json]: remove these spaces pls!

File: tokens/movistar.json [hunk 24]
```diff
@@ -125,26 +125,54 @@
...
       "description": "movistarBlue55"
     },
-    "buttonSecondaryBackground": {
+    "buttonSecondaryBorder": {
       "value": "{palette.movistarBlue}",
       "type": "color",
       "description": "movistarBlue"
     },
-    "buttonSecondaryBackgroundSelected": {
+
```
Comment (yceballost) [inline tokens/movistar.json]: remove these spaces pls!

File: tokens/blau.json [hunk 25]
```diff
@@ -140,11 +140,31 @@
...
+      "value": "{palette.blauBlueSecondary10}",
+      "type": "color",
+      "description": "blauBlueSecondary10"
+    },
     "buttonSecondaryBorderSelectedInverse": {
       "value": "{palette.blauBluePrimary30}",
       "type": "color",
       "description": "blauBluePrimary30"
     },
+    "buttonSecondaryBackgroundHoverInverse:": {
```
Comment (yceballost) [inline tokens/blau.json]:
```suggestion
    "buttonSecondaryBackgroundHoverInverse": {
```
same typo here

File: tokens/blau.json [hunk 26]
```diff
@@ -140,11 +140,31 @@
...
+      "value": "rgba({palette.blauBlueSecondary}, 0.2)",
+      "type": "color",
+      "description": "blauBlueSecondary"
+    },
+    "buttonSecondaryBackgroundSelectedInverse:": {
```
Comment (yceballost) [inline tokens/blau.json]:
```suggestion
    "buttonSecondaryBackgroundSelectedInverse": {
```
same typo here

**Review by yceballost — CHANGES_REQUESTED** [review]
movistar-classic have a lot of problems. Movistar classic will be an old skin, but we also need to define background hovers because still existing as a skin in Mística
image

File: tokens/movistar.json (hunk 23 above)
Comment (yceballost) [reply tokens/movistar.json]: still there :P

**Review by yceballost — CHANGES_REQUESTED** [review]
image
movistar-classic also need the new constants

File: tokens/o2.json [hunk 27]
```diff
@@ -125,26 +125,52 @@
...
       "description": "o2BluePrimary30"
     },
-    "buttonSecondaryBackground": {
+    "buttonSecondaryBorder": {
       "value": "{palette.o2BluePrimary}",
       "type": "color",
       "description": "o2BluePrimary"
     },
-    "buttonSecondaryBackgroundSelected": {
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 28]
```diff
@@ -125,26 +125,52 @@
...
+    "buttonSecondaryBorderSelected": {
       "value": "{palette.o2BluePrimary70}",
       "type": "color",
       "description": "o2BluePrimary70"
     },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 29]
```diff
@@ -125,26 +125,52 @@
...
+    "buttonSecondaryBackgroundHover": {
+      "value": "{palette.o2BluePrimary10}",
+      "type": "color",
+      "description": "o2BluePrimary10"
+    },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 30]
```diff
@@ -125,26 +125,52 @@
...
+    "buttonSecondaryBackgroundSelected": {
+      "value": "{palette.o2BluePrimary10}",
+      "type": "color",
+      "description": "o2BluePrimary10"
+    },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 31]
```diff
@@ -125,26 +125,52 @@
...
     "buttonSecondaryBorderInverse": {
       "value": "{palette.white}",
       "type": "color",
       "description": "white"
     },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 32]
```diff
@@ -125,26 +125,52 @@
...
     "buttonSecondaryBorderSelectedInverse": {
       "value": "{palette.o2BluePrimary30}",
       "type": "color",
       "description": "o2BluePrimary30"
     },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2.json [hunk 33]
```diff
@@ -180,6 +206,7 @@
       "type": "color",
       "description": "white"
     },
+
```
Comment (yceballost) [inline tokens/o2.json]: here

File: tokens/o2-classic.json [hunk 34]
```diff
@@ -125,26 +125,53 @@
...
+      "type": "color",
+      "description": "o2SkyBlueLight10"
+    },
+
     "buttonSecondaryBorderInverse": {
       "value": "{palette.white}",
       "type": "color",
       "description": "white"
     },
+
```
Comment (yceballost) [inline tokens/o2-classic.json]: here

File: tokens/o2-classic.json [hunk 35]
```diff
@@ -125,26 +125,53 @@
...
       "type": "color",
       "description": "o2SkyBlue55"
     },
+    
+    "buttonSecondaryBackgroundSelected": {
+      "value": "{palette.o2SkyBlueLight10}",
+      "type": "color",
+      "description": "o2SkyBlueLight10"
+    },
+
```
Comment (yceballost) [inline tokens/o2-classic.json]: here

File: tokens/o2-classic.json [hunk 36]
```diff
@@ -125,26 +125,53 @@
...
+      "type": "color",
+      "description": "o2SkyBlue55"
+    },
+
+    "buttonSecondaryBackgroundHover": {
       "value": "{palette.o2SkyBlue55}",
       "type": "color",
       "description": "o2SkyBlue55"
     },
+
```
Comment (yceballost) [inline tokens/o2-classic.json]: here

File: tokens/o2-classic.json [hunk 37]
```diff
@@ -125,26 +125,53 @@
...
       "description": "o2SkyBlueLight45"
     },
-    "buttonSecondaryBackground": {
+    "buttonSecondaryBorder": {
       "value": "{palette.o2SkyBlue}",
       "type": "color",
       "description": "o2SkyBlue"
     },
-    "buttonSecondaryBackgroundSelected": {
+
```
Comment (yceballost) [inline tokens/o2-classic.json]: here

**Review by yceballost — CHANGES_REQUESTED** [review]
looks great!😍 but there are still some empty spaces! just for neatness

File: tokens/vivo.json [hunk 38]
```diff
@@ -125,26 +125,53 @@
...
-    "buttonSecondaryBackgroundSelected": {
+  
+    "buttonSecondaryBorderSelected": {
+      "value": "{palette.vivoPurpleDark}",
+      "type": "color",
+      "description": "vivoPurpleDark"
+    },
+
+    "buttonSecondaryBackgroundHover": {
       "value": "{palette.vivoPurpleDark}",
```
Comment (yceballost) [inline tokens/vivo.json]: is it correct? maybe too dark for button background hover

Comment (AnaMontes11) [reply tokens/vivo.json]: Absolutely right, thanks!

File: tokens/o2.json (hunk 33 above)
Comment (yceballost) [reply tokens/o2.json]: this space is missing to remove

File: tokens/movistar-classic.json [hunk 39]
```diff
@@ -145,15 +155,25 @@
...
+      "description": "white"
+    },
     "textButtonPrimary": {
       "value": "{palette.white}",
       "type": "color",
       "description": "white"
     },
     "textButtonPrimaryInverse": {
-      "value": "{palette.movistarBlue}",
+      "value": "{palette.white}",
```
Comment (yceballost) [inline tokens/movistar-classic.json]: why did you change this?

if you use white with buttonPrimaryBackgroundInverse the button will be completly white!
```
 "buttonPrimaryBackgroundInverse": {
      "value": "{palette.white}",
      "type": "color",
      "description": "white"
    },
```

remember, movistar-classic have to work same as today, just include the new states for this skin

File: tokens/o2-classic.json [hunk 40]
```diff
@@ -145,6 +156,18 @@
       "type": "color",
       "description": "o2SkyBlueLight45"
     },
+
+    "buttonSecondaryBackgroundHoverInverse": {
+      "value": "rgba({palette.o2SkyBlueLight45}, 0.2)",
+      "type": "color",
+      "description": "o2SkyBlueLight45"
+    },
+
```
Comment (yceballost) [inline tokens/o2-classic.json]: remove this space

**Review by yceballost — CHANGES_REQUESTED** [review]

File: tokens/movistar-classic.json (hunk 39 above)
Comment (AnaMontes11) [reply tokens/movistar-classic.json]: fixed up

**Review by aweell — APPROVED** [review]
https://www.youtube.com/watch?v=3WAOxKOmR90

File: tokens/movistar.json [hunk 41]
```diff
@@ -573,14 +573,14 @@
       "description": "movistarBlue55"
     },
     "buttonSecondaryBackgroundHover": {
-      "value": "rgba({palette.movistarBlue}, 0.2)",
+      "value": "rgba({palette.white}, 0.15)",
```
Comment (yceballost) [inline tokens/movistar.json:576]: i don't understand this change, why do you use white in secondary backgroundHover?

Comment (yceballost) [reply tokens/movistar.json:576]: ok, dark mode XD

**Review by yceballost — APPROVED** [review]
